velox/functions/sparksql/aggregates/FirstLastAggregate.cpp

```diff
@@ -274,20 +364,31 @@ exec::AggregateRegistrationResult registerFirstLast(const std::string& name) {
       exec::AggregateFunctionSignatureBuilder()
           .typeVariable("T")
           .argumentType("T")
-          .intermediateType("T")
+          .intermediateType("row(T, boolean)")
           .returnType("T")
           .build()};
 
+  signatures.push_back(
```

Comment:
Let's split this PR into 2: (1) fix intermediate type; (2) add support for decimal inputs.

Reply:
Thanks, will split and modify PR title/description.